packages/graphql_server/src/chatGTPMethods.ts

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * The URL used to request the Response
+ */
+const postURL = 'https://api.openai.com/v1/chat/completions'
+
+/**
+ * Header for the request
+ */
+
+const applicationType = 'application/json'
+const openAiKey = 'Bearer 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'
+const errorMessage =
+  ' The fetched response was empty. Most likely there is something wrong with the JSON request. '
+/**
+ * Send a request to the ELI5 OpenAI API to generate a response.
+ * @param readME - The text to be explained.
+ * @returns The generated explanation from the ELI5 API.
+ */
+async function openAIRequestEli5(readME: string) {
+  const jsonRequestBody = JSON.parse(request_body_Eli5) as RequestBodyOpenAI
+  jsonRequestBody.messages[1].content = questionEli5
+  try {
+    const response = await axios.post(postURL, jsonRequestBody, {
+      headers: {
+        'Content-Type': applicationType,
+        Authorization: openAiKey
+      }
+    })
+
+    const data = response.data as ResponseBodyOpenAi
+    if (!data?.choices[0]?.message?.content) {
+      console.log(errorMessage)
+      return null
+    } else {
+      const content: string = data.choices[0].message.content
+      console.log(content)
+      return content
+    }
+  } catch (error) {
+    console.log('AI request did not work: ', error)
+    return null
+  }
+}
+
+/**
+ * Send a request to the Hackernews OpenAI API to generate a response.
+ * @param comments - The comments to be evaluated.
+ * @returns The generated response from the Hackernews API.
+ */
+async function openAIRequestHackernews(comments: string) {
+  const jsonRequestBody = JSON.parse(request_body_Hackernews) as RequestBodyOpenAI
+  jsonRequestBody.messages[1].content = questionHackernews + ' ' + comments
+
+  try {
+    const response = await axios.post(postURL, jsonRequestBody, {
+      headers: {
+        'Content-Type': applicationType,
+        Authorization: openAiKey
+      }
+    })
+
+    const data = response.data as ResponseBodyOpenAi
+    if (!data?.choices[0]?.message?.content) {
+      console.log(errorMessage)
+      return null
+    } else {
+      const content: string = data.choices[0].message.content
+      return content
+    }
+  } catch (error) {
+    console.log('OpenAI request did not work out: ', error)
+    return null
+  }
+}
+
+/**
+ * Send a request to the OpenAI API to generate a response that Categorizes the project.
+ * @param comments - The comments to be evaluated.
+ * @returns The generated response from the Hackernews API.
+ */
+
+async function categorizeProject(readMeOrCategory: string) {
+  const jsonRequestBody = JSON.parse(request_body_Categories) as RequestBodyOpenAI
+  const listOfCategories: string[] = [
```

Comment:
Please use the list of categories the fronted provides, which I also dm'ed you on slack, thanks.